src/Data/Foldable.purs

```diff
@@ -169,6 +170,10 @@ instance foldableMultiplicative :: Foldable Multiplicative where
 fold :: forall f m. Foldable f => Monoid m => f m -> m
 fold = foldMap id
 
+-- | Similar to 'foldl', but the result is encapsulated in a monad. 
```

Comment:
Of course, it could be stack-safe, if it were implemented using `List.fromFoldable`, but that's a different approach.